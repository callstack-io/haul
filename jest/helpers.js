/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

'use strict';

const traverse = require('traverse'); // eslint-disable-line
const path = require('path');
const excapeStringRegex = require('escape-string-regexp');

const flushPromises = (): Promise<any> =>
  new Promise(resolve => setTimeout(resolve));

const replacePathsInObject = (object: mixed) => {
  return traverse(object).map(
    entry =>
      typeof entry === 'string'
        ? entry
            .replace(/\/.*\/node_modules/, '<<NODE_MODULE>>')
            .replace(
              new RegExp(
                `^${excapeStringRegex(path.resolve(__dirname, '..'))}`
              ),
              '<<REPLACED>>'
            )
        : entry
  );
};

module.exports = {
  flushPromises,
  replacePathsInObject,
};
