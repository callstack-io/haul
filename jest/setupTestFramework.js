/* global jest */

jest.mock('chalk');

// Integration tests are quite slow to run on CI, feel free to adjust as needed
jest.setTimeout(80000);
