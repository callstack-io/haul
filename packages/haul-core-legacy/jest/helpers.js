/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

'use strict';

const traverse = require('traverse'); // eslint-disable-line
const path = require('path');
const excapeStringRegex = require('escape-string-regexp');

const flushPromises = (): Promise<any> =>
  new Promise(resolve => setTimeout(resolve));

const replacePathsInObject = (object: mixed) => {
  return traverse(object).map(
    entry =>
      typeof entry === 'string'
        ? entry
            .replace(
              new RegExp(`^${excapeStringRegex(path.resolve('/'))}`),
              '/'
            )
            .replace(new RegExp('\\\\', 'g'), '/')
            .replace(/\/.*\/node_modules/, '<<NODE_MODULE>>')
            .replace(
              new RegExp(
                process.platform === 'win32'
                  ? `^${excapeStringRegex(
                      path
                        .resolve(__dirname, '..')
                        .slice(2)
                        .replace(new RegExp('\\\\', 'g'), '/')
                    )}`
                  : `^${excapeStringRegex(path.resolve(__dirname, '..'))}`
              ),
              '<<REPLACED>>'
            )
        : entry
  );
};

const fixWorkers = (object: mixed) => {
  return traverse(object).map(entry => {
    if (typeof entry === 'object' && 'workers' in entry) {
      return Object.assign({}, entry, { workers: '<<CPUS>>' });
    }

    return entry;
  });
};

module.exports = {
  flushPromises,
  replacePathsInObject,
  fixWorkers,
};
