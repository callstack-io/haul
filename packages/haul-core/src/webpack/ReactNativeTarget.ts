import webpack from 'webpack';
import ReactNativeEnvPlugin from './plugins/ReactNativeEnvPlugin';

/**
 * // Based on Webpack's webworker target:
 * https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsApply.js#L84
 *
 * Webworker resembles the React Native JS execution environment the most with few
 * differences, which are tweaked with ReactNativeEnvPlugin.
 * Adding ReactNativeEnvPlugin directly to config's plugins list
 * does not result in correct bundle for some reason.
 */
export default function ReactNativeTarget(compiler: webpack.Compiler) {
  // Load modules using require, since we don't really care about types here.
  const WebWorkerTemplatePlugin = require('webpack/lib/webworker/WebWorkerTemplatePlugin');
  const NodeSourcePlugin = require('webpack/lib/node/NodeSourcePlugin');
  const FunctionModulePlugin = require('webpack/lib/FunctionModulePlugin');
  const LoaderTargetPlugin = require('webpack/lib/LoaderTargetPlugin');

  new WebWorkerTemplatePlugin().apply(compiler);
  new FunctionModulePlugin().apply(compiler);
  new NodeSourcePlugin(compiler.options.node).apply(compiler);
  new LoaderTargetPlugin(compiler.options.target).apply(compiler);

  // Apply React Native tweaks
  new ReactNativeEnvPlugin().apply(compiler);
}
