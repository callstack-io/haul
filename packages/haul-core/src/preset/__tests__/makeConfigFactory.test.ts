import webpack from 'webpack';
import makeConfigFactory from '../makeConfigFactory';
import { Runtime, ProjectConfig, EnvOptions } from '../../';
// @ts-ignore
import { replacePathsInObject } from 'jest/helpers'; // eslint-disable-line

const makeConfig = makeConfigFactory(
  (_runtime, env, bundleName, projectConfig) => ({
    mode: env.dev ? 'development' : 'production',
    context: env.root,
    devtool: false,
    entry: projectConfig.bundles[bundleName].entry,
    target: 'webworker',
    output: {
      path: env.assetsDest || env.root,
      publicPath: `http://${projectConfig.server.host}:${
        projectConfig.server.port
      }/`,
      globalObject: 'this',
    },
    plugins: [
      new webpack.DefinePlugin({
        'process.env': {
          NODE_ENV: env.dev ? '"development"' : '"production"',
        },
        __DEV__: env.dev,
      }),
      new webpack.LoaderOptionsPlugin({
        minimize: !!env.minify,
        debug: env.dev,
      }),
    ],
  })
);

function hasPlugin(webpackConfig: any, pluginName: string) {
  return webpackConfig.plugins.some(
    (p: any) => p.constructor.name === pluginName
  );
}

const runtime = new Runtime();

describe('makeConfig', () => {
  describe('with single bundle', () => {
    const baseEnv = {
      platform: 'ios',
      root: __dirname,
      dev: false,
      singleBundleMode: true,
    };

    it('should create config for basic-bundle', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(replacePathsInObject(config)).toMatchSnapshot();
    });

    it('should create config with multiple entry files', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: ['./a.js', './b.js'],
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.index.entry).toEqual(['./a.js', './b.js']);
    });

    it('should create config with builder function and custom transform', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: (env, runtime) => {
            expect(env.bundleType).toEqual('basic-bundle');
            expect(runtime).toBeDefined();

            return {
              entry: 'index.js',
              transform({ bundleName, config, runtime, env }) {
                expect(bundleName).toEqual('index');
                expect(runtime).toBeDefined();
                expect(env).toBeDefined();

                config.stats = 'minimal';
              },
            };
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.index.stats).toEqual('minimal');
    });

    it('should create config for indexed-ram-bundle', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'indexed-ram-bundle',
        bundle: true,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
    });

    it('should create config for file-ram-bundle', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'file-ram-bundle',
        bundle: true,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
    });

    it('should not create config with external source maps if sourceMap=false', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            sourceMap: false,
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'SourceMapDevToolPlugin')
      ).toBeFalsy();
    });

    it('should create config with inline source maps if sourceMap="inline"', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            sourceMap: 'inline',
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'EvalSourceMapDevToolPlugin')
      ).toBeTruthy();
    });

    it('should create config for packager server', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const runtime = new Runtime();
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'indexed-ram-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackRamBundlePlugin')
      ).toBeFalsy();
    });
  });

  describe('with multiple bundles', () => {
    const baseEnv = {
      platform: 'ios',
      root: __dirname,
      dev: false,
      singleBundleMode: false,
      bundle: true,
    };

    it('should create config', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: './host.js',
            dependsOn: ['base_dll'],
          },
          base_dll: {
            entry: './dll.js',
            dll: true,
            type: 'indexed-ram-bundle',
          },
          app: {
            entry: './spp.js',
            app: true,
            dependsOn: ['base_dll'],
            type: 'indexed-ram-bundle',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleOutput: __dirname,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.base_dll, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.app, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
      expect(replacePathsInObject(config)).toMatchSnapshot();
    });

    it('should create config for packager server', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: './host.js',
            dependsOn: ['base_dll'],
          },
          base_dll: {
            entry: './dll.js',
            dll: true,
            type: 'indexed-ram-bundle',
          },
          app: {
            entry: './spp.js',
            app: true,
            dependsOn: ['base_dll'],
            type: 'indexed-ram-bundle',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundle: false,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.base_dll, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.app, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(replacePathsInObject(config)).toMatchSnapshot();
    });
  });
});
