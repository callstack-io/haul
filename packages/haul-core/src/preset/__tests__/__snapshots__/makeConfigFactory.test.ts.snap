// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeConfig with multiple bundles should create config 1`] = `
Object {
  "bundles": Object {
    "app": Object {
      "app": true,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Object {
        "entryFiles": Array [
          "./spp.js",
        ],
        "setupFiles": Array [],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "app",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "indexed-ram-bundle",
    },
    "base_dll": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [],
      "dev": false,
      "dll": true,
      "entry": Object {
        "entryFiles": Array [
          "./dll.js",
        ],
        "setupFiles": Array [],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "base_dll",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "indexed-ram-bundle",
    },
    "index": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Object {
        "entryFiles": Array [
          "./host.js",
        ],
        "setupFiles": Array [],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "index",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
  },
  "platforms": Array [
    "ios",
    "android",
  ],
  "server": Object {
    "host": "localhost",
    "port": 8081,
  },
  "templates": Object {
    "filename": Object {
      "__server__": "[bundleName].[platform].bundle",
      "android": "[bundleName].[platform].bundle",
      "ios": "[bundleName].jsbundle",
    },
  },
  "webpackConfigs": Object {
    "app": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "./spp.js",
        ],
        "setupFiles": Array [],
      },
      "mode": "production",
      "output": Object {
        "filename": "app.jsbundle",
        "globalObject": "this",
        "library": "app",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        WebpackRamBundlePlugin {
          "bundleId": "app",
          "bundleName": "app",
          "indexRamBundle": true,
          "maxWorkers": 7,
          "minify": false,
          "minifyOptions": undefined,
          "modules": Array [],
          "name": "WebpackRamBundlePlugin",
          "preloadBundles": Array [
            "base_dll",
          ],
          "singleBundleMode": false,
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0,\\"base_dll\\":1,\\"app\\":2}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
    "base_dll": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "./dll.js",
        ],
        "setupFiles": Array [],
      },
      "mode": "production",
      "output": Object {
        "filename": "base_dll.jsbundle",
        "globalObject": "this",
        "library": "base_dll",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        WebpackRamBundlePlugin {
          "bundleId": "base_dll",
          "bundleName": "base_dll",
          "indexRamBundle": true,
          "maxWorkers": 7,
          "minify": false,
          "minifyOptions": undefined,
          "modules": Array [],
          "name": "WebpackRamBundlePlugin",
          "preloadBundles": Array [],
          "singleBundleMode": false,
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0,\\"base_dll\\":1,\\"app\\":2}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
        DllPlugin {
          "options": Object {
            "name": "base_dll",
            "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
          },
        },
      ],
      "target": "webworker",
    },
    "index": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "./host.js",
        ],
        "setupFiles": Array [],
      },
      "mode": "production",
      "output": Object {
        "filename": "index.jsbundle",
        "globalObject": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": true,
          "preloadBundles": Array [
            "base_dll",
          ],
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0,\\"base_dll\\":1,\\"app\\":2}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
  },
}
`;

exports[`makeConfig with multiple bundles should create config for packager server 1`] = `
Object {
  "bundles": Object {
    "app": Object {
      "app": true,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Object {
        "entryFiles": Array [
          "./spp.js",
        ],
        "setupFiles": Array [],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "app",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
    "base_dll": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [],
      "dev": false,
      "dll": true,
      "entry": Object {
        "entryFiles": Array [
          "./dll.js",
        ],
        "setupFiles": Array [],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "base_dll",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
    "index": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Object {
        "entryFiles": Array [
          "./host.js",
        ],
        "setupFiles": Array [],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "index",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
  },
  "platforms": Array [
    "ios",
    "android",
  ],
  "server": Object {
    "host": "localhost",
    "port": 8081,
  },
  "templates": Object {
    "filename": Object {
      "__server__": "[bundleName].[platform].bundle",
      "android": "[bundleName].[platform].bundle",
      "ios": "[bundleName].jsbundle",
    },
  },
  "webpackConfigs": Object {
    "app": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "./spp.js",
        ],
        "setupFiles": Array [],
      },
      "mode": "production",
      "output": Object {
        "filename": "app.ios.bundle",
        "globalObject": "this",
        "library": "app",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [
            "base_dll",
          ],
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0,\\"base_dll\\":1,\\"app\\":2}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
    "base_dll": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "./dll.js",
        ],
        "setupFiles": Array [],
      },
      "mode": "production",
      "output": Object {
        "filename": "base_dll.ios.bundle",
        "globalObject": "this",
        "library": "base_dll",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [],
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0,\\"base_dll\\":1,\\"app\\":2}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
        DllPlugin {
          "options": Object {
            "name": "base_dll",
            "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
          },
        },
      ],
      "target": "webworker",
    },
    "index": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "./host.js",
        ],
        "setupFiles": Array [],
      },
      "mode": "production",
      "output": Object {
        "filename": "index.ios.bundle",
        "globalObject": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [
            "base_dll",
          ],
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0,\\"base_dll\\":1,\\"app\\":2}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
  },
}
`;

exports[`makeConfig with single bundle should create config for basic-bundle 1`] = `
Object {
  "bundles": Object {
    "index": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [],
      "dev": false,
      "dll": false,
      "entry": Object {
        "entryFiles": Array [
          "react-native/Libraries/Core/InitializeCore.js",
          "index.js",
        ],
        "setupFiles": Array [
          "react-native/Libraries/Core/InitializeCore.js",
        ],
      },
      "external": false,
      "hasteOptions": Object {},
      "looseMode": [Function],
      "maxWorkers": 7,
      "minify": false,
      "minifyOptions": undefined,
      "name": "index",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
  },
  "platforms": Array [
    "ios",
    "android",
  ],
  "server": Object {
    "host": "localhost",
    "port": 8081,
  },
  "templates": Object {
    "filename": Object {
      "__server__": "[bundleName].[platform].bundle",
      "android": "[bundleName].[platform].bundle",
      "ios": "[bundleName].jsbundle",
    },
  },
  "webpackConfigs": Object {
    "index": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Object {
        "entryFiles": Array [
          "react-native/Libraries/Core/InitializeCore.js",
          "index.js",
        ],
        "setupFiles": Array [
          "react-native/Libraries/Core/InitializeCore.js",
        ],
      },
      "mode": "production",
      "output": Object {
        "filename": "index.jsbundle",
        "globalObject": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
        "sourceMapFilename": "[file].map",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [],
          "sourceMap": true,
        },
        DefinePlugin {
          "definitions": Object {
            "process.env.HAUL_BUNDLES": "{\\"index\\":0}",
          },
        },
        LooseModeWebpackPlugin {
          "checkLooseMode": [Function],
        },
      ],
      "target": "webworker",
    },
  },
}
`;
